scalems/wrappers/__init__.py

```diff
@@ -16,32 +15,109 @@
+        well in the keyword format.  We could, for now, just replace
+        the file containing the molecular system definition entirely
+        using a modify_input.  This might be the best bet for now, but
+        we will want to think about what sorts of changes might be of
+        interest, if any in the future.
+
+        The issue is that in most cases, if the chemical system is
+        changing, it usually requires changes of many lines of
+        topologies, and so passing it in as a dictionary will likely
+        be a big pain. Likely easier to take care of it outside the
+        simulation for now. Eventually, if some parallel work on
+        simulation topology definitions in Open Force Field works,
+        then it may be possible to do this better.
+
+        Things like "training potential" that get attached in the
+        examples should probably be attached at this point in the
+        workflow, rather than at the simulate. For most workflows, it
+        could ONLY be added at this stage, since potentials are
+        defined in the input files.  For GMX_API driven simulations,
+        the actual attachment of the potentials to the md simulation
+        could happen in the simulate function, but it would be better
+        if they were loaded into the data structure in the
+        modify_input sage.
+
     simulate(simulation_input: SimulationInput, **kwargs) -> Simulation:
         Create a simulation operation instance. Accepts input from *make_input*
         or *modify_input*, along with key word arguments specified by the wrapper
         module.
 
+        Potentially, we use this to specify which resources assoicated
+        with the job?  The goal of this function is to define a task
+        that is ready to dispatch for execution.  
+
+        MRS: some notes that I took during the conversation last week:
+        Any of the things that affect the checkpointable identity are
+        immutable after simulate.  So after simulate, define an
+        equivalence class. Operate by fingerprinting commands and data
+        to determine identity that should distinguish it.  Workflow
+        level checkpointing.
+       
+        MRS: I'm still a little confused by the point of this.  I
+        assume the importance of checkpointing is to know whether one
+        should relaunch a simulation if it fails?  And if it's not
+        checkpointable?  Can we clarify this a bit more.
+
+
 Utility functions:
     internal_to_pdb(structure) -> scalems.FileFuture:
         For interoperability with other tools, simulator wrappers should provide
         conversion utilities for native structure and topology data. *internal_to_pdb*
         converts a native structure object to a structure file in the Protein Data Bank
-        (PDB) format.
+        (PDB) format.  
+
+        MRS: Though actually, PDB might not be the best, since it's limited resolution. 
```

Comment:
PDB is non-ideal, but it *is* the most widely used molecular structure format.  I think it's a good place to live for the moment.  (numpy array / etc loses all the atom information)

Reply:
That's why I said something like mdtraj or mdanalysis object, so that atomic information is kept.  Yes, I am arguing that information should be kept. PDB is pretty lossy.  To me it makes more sense to convert the different files to a common internal object, rather than a lossy filetype.  I guess there is some question if it needs to be written out for checkpointing that it might need to be written out to file.